### PR TITLE
Fix explain InsertStmt NPE in FOLLOWER node

### PR DESCRIPTION
**Pull request by kangkaisen**
For https://github.com/apache/incubator-doris/issues/2096

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/analysis/InsertStmt.java [hunk 1]
```diff
@@ -762,4 +762,13 @@ public void reset() {
         dataPartition = null;
         targetColumns.clear();
     }
+
+    @Override
+    public RedirectStatus getRedirectStatus() {
+        if (isExplain()) {
+            return null;
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/analysis/InsertStmt.java]: Why not return `NO_FORWARD`?

File: fe/src/main/java/org/apache/doris/analysis/InsertStmt.java [hunk 2]
```diff
@@ -762,4 +762,13 @@ public void reset() {
...
+        } else {
+            return RedirectStatus.FORWARD_WITH_SYNC;
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/analysis/InsertStmt.java]: I think `FORWARD_NO_SYNC` is enough. Explain Insert does not modify any meta data that need to be sync.

**Review by morningman — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/analysis/InsertStmt.java (hunk 1 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/analysis/InsertStmt.java]: I referred  to here:
```
        if (redirectStatus == null) {
            return false;
        } else {
            return redirectStatus.isForwardToMaster();
        }
```
OK. I will change to `NO_FORWARD`.

File: fe/src/main/java/org/apache/doris/analysis/InsertStmt.java (hunk 2 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/analysis/InsertStmt.java]: `RedirectStatus.FORWARD_WITH_SYNC` is for insert, not  Explain Insert. So We should use `FORWARD_WITH_SYNC`

File: fe/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 3]
```diff
@@ -585,6 +585,11 @@ private void handleInsertStmt() throws Exception {
         } else {
             insertStmt = (InsertStmt) parsedStmt;
         }
+
+        // assign request_id
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/qe/StmtExecutor.java]: Better to comment why you assign query id here.

File: fe/src/main/java/org/apache/doris/analysis/InsertStmt.java (hunk 2 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/analysis/InsertStmt.java]: O, i see

**Review by morningman — DISMISSED** [review]

File: fe/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 3 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/qe/StmtExecutor.java]: OK, done.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**